### PR TITLE
feat: implement INDIRECT function

### PR DESCRIPTION
**Pull request by jafin**
## Summary
- Implement the `INDIRECT(ref_text, [a1])` Excel function that converts text strings into cell references at evaluation time
- Supports A1 style (default), absolute R1C1 (`R1C1`, `R1C1:R5C3`), sheet prefixes (`Sheet2!A1`), range references, and defined name resolution
- Registered as volatile (`FunctionFlags.Volatile`) to ensure recalculation, with dynamic dependency handling in `DependenciesVisitor`

## Changes
- **`Lookup.cs`** — Added `Indirect`, `TryParseR1C1Reference`, `TryResolveDefinedName`, `EvaluateDefinedName` methods; registered function
- **`SignatureAdapter.cs`** — Changed `ToText` and `CoerceToLogical` from `private` to `internal` for reuse by raw Span-based functions
- **`DependenciesVisitor.cs`** — Added INDIRECT case (dynamic refs can't be statically resolved; volatile flag handles recalc)
- **`FormulaParserTests.cs`** — Uncommented existing `INDIRECT("A2")` test case
- **`LookupTests.cs`** — Added 11 tests covering basic A1, absolute refs, ranges, empty string, sheet prefix, invalid ref, defined names, A1 flag, R1C1, nonexistent sheet, dynamic reference

## Test plan
- [x] All 11 new INDIRECT tests pass on net8.0 and net10.0
- [x] Uncommented `Ref_function` parser test passes
- [x] Full test suite (6007 tests) passes with zero regressions on both TFMs
- [ ] Verify R1C1 range references work (e.g., `INDIRECT("R1C1:R5C3", FALSE)`)
- [ ] Verify edge cases with quoted sheet names (e.g., `'My Sheet'!A1`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * INDIREC T support added: resolves text references to cells/ranges with A1 and R1C1 styles, absolute/range and sheet-qualified references, and defined-name resolution; treated as volatile with dependency tracking.

* **Tests**
  * Re-enabled and expanded unit tests exercising INDIRECT across many formats and edge cases (including invalid or missing references).

### CODE REVIEW

File: XLibur/Excel/CalcEngine/Functions/Lookup.cs [hunk 1]
```diff
@@ -680,4 +681,128 @@ private static AnyValue RowsOrColumns(AnyValue value, bool rows)
...
+
+        if (firstRow < 1 || firstCol < 1 || lastRow < 1 || lastCol < 1
+            || firstRow > XLHelper.MaxRowNumber || firstCol > XLHelper.MaxColumnNumber
+            || lastRow > XLHelper.MaxRowNumber || lastCol > XLHelper.MaxColumnNumber)
+            return XLError.CellReference;
+
+        var firstAddress = new XLAddress(worksheet, firstRow, firstCol, true, true);
+        var lastAddress = new XLAddress(lastRow, lastCol, true, true);
+        var rangeAddress = new XLRangeAddress(firstAddress, lastAddress);
+        return new Reference(rangeAddress.Normalize());
```
[the remark this answers is not in the document]
Comment (jafin) [reply XLibur/Excel/CalcEngine/Functions/Lookup.cs:786]: This finding is a false positive. The current code follows the same pattern used throughout the codebase
  (FromSheetRange, Intersect, etc.). The XLRangeAddress.Worksheet is set from firstAddress.Worksheet, so the returned
  Reference has correct worksheet context. No fix needed.
